src/plugins/intel_gpu/src/runtime/layout.cpp

```diff
@@ -603,6 +603,29 @@ static inline bool check_redundant_1d_along_feature(layout const& l1, layout con
             return true;
         }
 
+        if (max_inner_blk > 1) {
+            auto l1_inner_blk_idx = format::is_single_blocked(l1.format) ? l1.format.traits().block_sizes.at(0).first : SIZE_MAX;
+            auto l2_inner_blk_idx = format::is_single_blocked(l2.format) ? l2.format.traits().block_sizes.at(0).first : SIZE_MAX;
+            auto min_inner_blk_idx = std::min(l1_inner_blk_idx, l2_inner_blk_idx);
+            auto is_scalar_below_block_idx = [](layout const& l, size_t block_idx) {
+                if (block_idx >= l.get_rank())
+                    return false;
+                int32_t total = 1;
+                for (size_t i = block_idx + 1; i < l.get_rank(); ++i) {
+                    total *= l.get_dim(i);
+                }
+                if (total == 1)
+                    return true;
+                else
+                    return false;
+            };
+            if (l1.batch() == l2.batch() &&
+                is_scalar_below_block_idx(l1, min_inner_blk_idx) && is_scalar_below_block_idx(l2, min_inner_blk_idx) &&
+                l2.feature() == l1.feature() && (l2.feature() % max_inner_blk == 0)) {
+                return true;
+            }
```

Comment:
I think this is very difficult to understand. Let's discuss offline.

Reply:
As we discussed offline, I updated it. No ambiguity and reasonable.